### PR TITLE
Teams created by events should not receive team notifications

### PR DESCRIPTION
**Pull request by lheilmann**
See #494

### CODE REVIEW

File: app/models/team.rb [hunk 1]
```diff
@@ -51,6 +52,10 @@ def has_member?(user)
     members.include?(user)
   end
 
+  def is_qualified_to_receive_notifications?
+    !self.created_by_event?
```
Comment (bakoe) [inline app/models/team.rb]: Any reason for not using `self.created_by_user?` here? To my mind, this would increase the readability by avoiding the negation.

File: app/models/team_user.rb [hunk 2]
```diff
@@ -34,6 +34,9 @@ def delete_ownership
 
   private
     def send_mail_when_created
-      TeamMailer.user_added_to_team(User.find(user_id), Team.find(team_id)).deliver_now
+      team = Team.find(team_id)
+      if team.is_qualified_to_receive_notifications?
```
Comment (bakoe) [inline app/models/team_user.rb:38]: Nice!

File: db/schema.rb [hunk 3]
```diff
@@ -37,8 +37,8 @@
     t.integer "points_for_win", default: 3
     t.integer "points_for_draw", default: 1
     t.integer "points_for_lose", default: 0
-    t.boolean "has_place_3_match", default: true
     t.text "image_data"
+    t.boolean "has_place_3_match", default: true
```
Comment (bakoe) [inline db/schema.rb:41]: You could discard/revert this change as was well the one from line 154/155. I don’t know why this always happens—the same re-ordering of existing columns in the Schema happens on my machine as well when running newly created migrations.

File: spec/models/team_spec.rb [hunk 4]
```diff
@@ -102,6 +102,13 @@
     expect { team.members << user }.to change { ActionMailer::Base.deliveries.length }.by(1)
   end
 
+  it 'should not notify newly added team members when team is a single' do
```
Comment (bakoe) [inline spec/models/team_spec.rb]: You could change the „single“ wording into „created by event“ here as well to avoid confusion.

**Review by bakoe — COMMENTED** [review]
Some minor comments – however, I really like the changes. Haven’t tested in Browser though.

File: app/models/team.rb (hunk 1 above)
Comment (lheilmann) [reply app/models/team.rb]: I added `created_by_user` later... But you're definitely right, I'll change it!

File: spec/models/team_spec.rb (hunk 4 above)
Comment (lheilmann) [reply spec/models/team_spec.rb]: :+1:

File: app/models/team.rb (hunk 1 above)
Comment (lheilmann) [reply app/models/team.rb]: Well, I just recognized that `created_by_user` is a scope for `Team` and no instance variable. So `self.created_by_user?` will result in lots of errors. I will revert to the original `!self.created_by_event?` or do you have any other ideas?

File: app/models/user.rb [hunk 5]
```diff
@@ -78,12 +78,12 @@ def four_different_characters
 
   has_many :team_users
   has_many :teams, through: :team_users, source: :team
-  has_many :non_single_teams, -> { where single: false }, through: :team_users, source: :team
+  has_many :teams_created_by_event, -> { where created_by_event: false }, through: :team_users, source: :team
```
Comment (niconomaa) [inline app/models/user.rb]: @lheilmann can you explain this line to me?
at first sight, it seems paradoxical to me that this lines reads `where created_by_event: false`. i might be missing something here, but that might also indicate that readibility could be improved?

File: app/models/team.rb (hunk 1 above)
Comment (jonasnoki) [reply app/models/team.rb]: I would agree with @lheilmann here. The only way to use created_by_user would be to intersect with the scope... not very useful.

Comment (bakoe) [reply app/models/team.rb]: Aah, okay – sorry, I didn't notice that `created_by_user` is only a scope and not an instance variable. Then it's fine to leave it at the original `!self.created_by_event?`. 👍

File: db/schema.rb (hunk 3 above)
Comment (lheilmann) [reply db/schema.rb:41]: @bakoe Is there a quick and easy way to do what you recommend? Otherwise I would suggest to just let rails be weird in this case... 🤪

File: app/models/team.rb (hunk 1 above)
Comment (niconomaa) [reply app/models/team.rb]: if this is purely about readability, what you could do is use `not` instead of `!` for your negation here. the operators are slightly different in precedence, but should both work here...

File: db/schema.rb (hunk 3 above)
Comment (bakoe) [reply db/schema.rb:41]: @lheilmann I can think of nothing quicker than manually adding a commit that changes `schema.rb` to be exactly the same as the one on `dev` with only the renamed field you introduced as difference. This would lead to a diff of this PR that would only contain you re-naming but not the apparent re-ordering of existing database columns.

However, I think it's pretty irrelevant – furthermore, we can't know for sure that the next person running another migration would not run into these re-ordering “issues” again. Thus, I would suggest just leaving the `schema.rb` changes in your PR as they are at the moment.

Comment (bakoe) [reply db/schema.rb:41]: Maybe it's even an anti-pattern to manually change the `schema.rb`, so leaving it as it is now (created by `db:migrate`) is probably the best practice.

File: app/models/team.rb (hunk 1 above)
Comment (lheilmann) [reply app/models/team.rb]: Good idea! :+1:

File: db/schema.rb (hunk 3 above)
Comment (lheilmann) [reply db/schema.rb:41]: Alright!

File: app/models/user.rb (hunk 5 above)
Comment (bakoe) [reply app/models/user.rb]: You're right @tomkellog – thanks for noticing! I didn't notice before but the variable is actually named wrongly. Only the “real” teams should be displayed on the Teams index page (see #488). The current variable name implies that the opposite—the ”fake“ teams created by events—is displayed on the Teams index page (see `/views/teams/index.html.erb`).

@lheilmann Please change the variable name to something like `teams_not_created_by_event`. Thanks!

Comment (lheilmann) [reply app/models/user.rb]: You're both right - thanks for noticing! Already changed to `created_by_user` and refactored.

File: app/models/user.rb [hunk 6]
```diff
@@ -78,12 +78,13 @@ def four_different_characters
 
   has_many :team_users
   has_many :teams, through: :team_users, source: :team
-  has_many :non_single_teams, -> { where single: false }, through: :team_users, source: :team
+  has_many :teams_created_by_user, -> { where created_by_event: false }, through: :team_users, source: :team
+  has_many :teams_created_by_event, -> { where created_by_event: true }, through: :team_users, source: :team
```
Comment (bakoe) [inline app/models/user.rb]: Is `:teams_created_by_event` used anywhere? If not, it could be removed for the moment imho.

**Review by bakoe — APPROVED** [review]

**Review by pawelsoundso — APPROVED** [review]
